### ISSUE TITLE
Silence GTKSharp Warnings

### ISSUE THREAD

**Issue by turbo** [picoe/Eto#1211]
With Eto.Gtk, GtkSharp spews a lot of errors, warnings and "critical" messages into the console. However, I have yet to encounter a "critical" GTK error that actually affects the UI :). Is there a way to silence Gtk's output?

Example noise:

```text
[WARNING] Function not found: gtk_normalise_detailed_action_name
[WARNING] Function not found: gtk_show_uri_on_window
[WARNING] Function not found: gtk_widget_get_focus_on_click
[WARNING] Function not found: gtk_widget_set_focus_on_click
[WARNING] Function not found: gtk_widget_get_allocated_size
[WARNING] Function not found: gtk_widget_queue_allocate
[WARNING] Function not found: gtk_scrolled_window_get_max_content_width
[WARNING] Function not found: gtk_scrolled_window_set_max_content_width
[WARNING] Function not found: gtk_scrolled_window_get_max_content_height
[WARNING] Function not found: gtk_scrolled_window_set_max_content_height
[WARNING] Function not found: gtk_scrolled_window_get_propagate_natural_width
[WARNING] Function not found: gtk_scrolled_window_set_propagate_natural_width
[WARNING] Function not found: gtk_scrolled_window_get_propagate_natural_height
[WARNING] Function not found: gtk_scrolled_window_set_propagate_natural_height
[WARNING] Function not found: gdk_pixbuf_new_from_file_at_scale_utf8
[WARNING] Function not found: gdk_pixbuf_new_from_file_at_size_utf8
[WARNING] Function not found: gdk_pixbuf_new_from_file_utf8
[WARNING] Function not found: gdk_pixbuf_copy_options
[WARNING] Function not found: gdk_pixbuf_remove_option
[WARNING] Function not found: gdk_pixbuf_save_to_streamv
[WARNING] Function not found: gdk_pixbuf_save_to_streamv_async
[WARNING] Function not found: gdk_pixbuf_savev_utf8
[WARNING] Function not found: gtk_text_view_reset_cursor_blink
[WARNING] Function not found: gtk_menu_place_on_monitor
[WARNING] Function not found: gtk_menu_popup_at_pointer
[WARNING] Function not found: gtk_menu_popup_at_rect
[WARNING] Function not found: gtk_menu_popup_at_widget
[WARNING] Function not found: gdk_window_begin_draw_frame
[WARNING] Function not found: gdk_window_end_draw_frame

(Linux:6434): Gtk-CRITICAL **: gtk_widget_set_allocation: assertion '_gtk_widget_get_visible (widget) || _gtk_widget_is_toplevel (widget)' failed

(Linux:6434): Gtk-CRITICAL **: gtk_widget_set_allocation: assertion '_gtk_widget_get_visible (widget) || _gtk_widget_is_toplevel (widget)' failed

(Linux:6434): Gtk-CRITICAL **: gtk_widget_set_allocation: assertion '_gtk_widget_get_visible (widget) || _gtk_widget_is_toplevel (widget)' failed

(Linux:6434): Gtk-CRITICAL **: gtk_widget_set_allocation: assertion '_gtk_widget_get_visible (widget) || _gtk_widget_is_toplevel (widget)' failed
[WARNING] Function not found: g_source_get_type
[WARNING] Function not found: gtk_text_iter_starts_tag
[WARNING] Function not found: gtk_style_context_to_string
[WARNING] Function not found: cairo_atsui_font_face_create_for_atsu_font_id
[WARNING] Function not found: cairo_glitz_surface_create
[WARNING] Function not found: cairo_quartz_surface_create
[WARNING] Function not found: cairo_directfb_surface_create
[WARNING] Function not found: cairo_win32_font_face_create_for_logfontw
[WARNING] Function not found: cairo_win32_scaled_font_done_font
[WARNING] Function not found: cairo_win32_scaled_font_get_metrics_factor
[WARNING] Function not found: cairo_win32_scaled_font_select_font
[WARNING] Function not found: cairo_win32_surface_create
[WARNING] Function not found: cairo_win32_surface_create_with_ddb
[WARNING] Function not found: gtk_render_background_get_clip
[WARNING] Function not found: gdk_display_get_default_seat
[WARNING] Function not found: gdk_display_get_monitor
[WARNING] Function not found: gdk_display_get_monitor_at_point
[WARNING] Function not found: gdk_display_get_monitor_at_window
[WARNING] Function not found: gdk_display_get_n_monitors
[WARNING] Function not found: gdk_display_get_primary_monitor
[WARNING] Function not found: gdk_display_list_seats
```

**Comment by cwensley** [comment]
This might be a good question to post on https://github.com/GtkSharp/GtkSharp, the warning output isn't something Eto is doing.

However, the critical errors should be fixed if possible, even though things seem to work.  Do you have some code that can reproduce them?  It would also help to know what OS and version you are running it on.

**Comment by turbo** [comment]
I've just been clicking through the Eto.Test project with the GtkSharp target using the Mono 5.14 targets on Ubuntu 16.04 and 18.04 based systems. (Eto isn't really unique in this, Gtk apps tend to output a lot of allegedly critical errors.)

**Comment by turbo** [comment]
Reference: https://github.com/GtkSharp/GtkSharp/issues/18

**Comment by turbo** [comment]
So it's fixed in GtkSharp 3.22.24.30 (apparently). Does Eto use an older version (I'm using the myget feed)?

**Comment by turbo** [comment]
To answer this: Yes. Eto should update it's pkg reference to the latest GtkSharp. Fixes a whole lot of warnings.
